### PR TITLE
Response requisition modal part 02

### LINKED ISSUE #491: Response Requisition Modal
Response requisition charts have 2 main objectives:
- tell the supplying store if he has enough stock to meet customer's requested quantity as well as requested quantities from all other customers. 
- tell the supplying store about the current supply situation at customer's

These are just ideas but I think the tab 1 would be a nice minimum: 

Tab 1 (My store): 
Using same kind of charts as for the request requisition, display
- the SoH of the supplying store (and why not in the future what is in transit or on order)
- the requested quantity of the customer + sum of requested quantities from all other customer
The objective is to be able to make the best choice when allocating a quantity. In the mockup, supplying store doesn't have enough to cover all the demand. He may need to allocate less than the customer's requested quantity

Response Requisition Modal (1)

Tab 2 (Customer)
The same chart visible by the customer so that the supplying store can assess the current situation of the customer. 

Response Requisition Modal (2)

Another thing would be for the system to automatically calculate the quantity to allocate when there is not enough stock:
- Requested Quantity: Customer A = 25, Customer B = 35, Customer C = 40 (Total = 100)
- Supplier's Stock on Hand = 50
- Customer's AMC: Customer A = 10, Customer B = 20, Customer C = 50
- System would then allocate the 50 units based on the AMC of each store: 
-- Customer A = 6
-- Customer B = 13
-- Customer C = 31

This way, we help user decide a fair way of distributing stock when there is not enough ! (Just an idea) System could suggest it the same way we are suggesting quantity to order (with a button ?). 
Another thing you usually do is to define a minimum stock that you keep for yourself so you don't give away everything.

### PR DESCRIPTION
**Pull request by roxy-dao**
Closes #491 

#### My Store
- Shows predicted total stock (soh + incoming stock + stock on order)
- Show total requests for stock

#### Customer 
- Monthly consumption based on MOS and average consumption

Bars look a bit funny when the values of the two bars have a huge difference.... 👀 

Second part of the `Customer` tab pending for now...
Screen Shot 2022-11-25 at 9 00 36 AM

### CODE REVIEW

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  otherRequestedQuantity,
+}) => {
+  const t = useTranslation('replenishment');
+  const predictedStockLevels = stockOnHand + incomingStock + stockOnOrder;
+  const totalRequested = requestedQuantity + otherRequestedQuantity;
+
+  const predictedStockPercent =
+    predictedStockLevels < totalRequested
+      ? Math.round((100 * predictedStockLevels) / totalRequested).toString() +
+        '%'
```
Comment (roxy-dao) [inline client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx:81]: Bars would adjust themselves properly without the '%' 👀

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ResponseLineEdit/ResponseLineEditForm.tsx [hunk 2]
```diff
@@ -138,7 +138,7 @@ export const ResponseLineEditForm = ({
             {t('heading.comment')}
           </Typography>
           <TextArea
-            value={draftLine.comment}
+            value={draftLine.comment ?? ''}
```
Comment (roxy-dao) [inline client/packages/requisitions/src/ResponseRequisition/DetailView/ResponseLineEdit/ResponseLineEditForm.tsx:141]: Cleared a console warning

File: client/packages/common/src/ui/components/navigation/Tabs/ModalTab.tsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ const StyledTabPanel = styled(TabPanel)({
 
 const StyledTabContainer = styled(Box)(({ theme }) => ({
   borderColor: theme.palette.divider,
-  flexDirection: 'row',
+  flexDirection: 'column',
```
Comment (mark-prins) [inline client/packages/common/src/ui/components/navigation/Tabs/ModalTab.tsx:18]: I was nervous about the impact on other uses of the ModalTab.. but is fine so I'll calm down.

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/RequestStoreStats.tsx [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+const MIN_FLEX_BASIS_TO_SHOW_LABEL = 10;
+const MIN_FLEX_BASIS_TO_SHOW_VALUE = 5;
+const MIN_MC_WIDTH_TO_SHOW_TEXT = 5;
+
+const Divider = () => (
+  <Box sx={{ backgroundColor: 'gray.dark', width: '1px', height: '45px' }} />
+);
+
+const ValueBar = ({
```
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/RequestStoreStats.tsx:25]: I've made #873 to lift this out to a common component

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/RequestStoreStats.tsx [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+  const t = useTranslation('replenishment');
+  const detail = isAmcZero
+    ? `: ${t('error.amc-is-zero')}`
+    : isSohAndQtyZero
+    ? `: ${t('error.soh-and-suggested-quantity-are-zero')}`
+    : '';
+  const message = `${t('error.unable-to-calculate')}${detail}`;
+
+  return (
+    <Box display="flex" padding={1} gap={1}>
```
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/RequestStoreStats.tsx:149]:
```suggestion
    <Box display="flex" padding={2} gap={1}>
```

The error is jammed up against the edge.. a little more padding would be good

Screenshot 2022-11-30 at 9 25 48 AM

File: client/packages/requisitions/src/ResponseRequisition/api/hooks/line/useResponseLineStatsData.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { useQuery } from '@openmsupply-client/common';
+import { useResponseApi } from '../utils/useResponseApi';
+
+export const useRequestLineStatsData = (requisitionLineId: string) => {
```
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/api/hooks/line/useResponseLineStatsData.ts:4]: Should this be:
```suggestion
export const useResponseLineStatsData = (requisitionLineId: string) => {
```

File: client/packages/requisitions/src/ResponseRequisition/api/hooks/line/index.ts [hunk 7]
```diff
@@ -1,9 +1,11 @@
 import { useDeleteResponseLines } from './useDeleteResponseLines';
 import { useResponseLines } from './useResponseLines';
+import { useRequestLineStatsData } from './useResponseLineStatsData';
```
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/api/hooks/line/index.ts:3]: I think this should be `useResponseLineStatsData` ?

File: client/packages/requisitions/src/ResponseRequisition/api/hooks/index.ts [hunk 8]
```diff
@@ -14,6 +14,7 @@ export const useResponse = {
   },
   line: {
     list: Lines.useResponseLines,
+    stats: Lines.useRequestLineStatsData,
```
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/api/hooks/index.ts:17]:
```suggestion
    stats: Lines.useResponseLineStatsData,
```
??

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ResponseLineEdit/ResponseLineEdit.tsx [hunk 9]
```diff
@@ -26,6 +29,29 @@ export const ResponseLineEdit = ({
...
+  const tabs = [
+    {
+      Component: <ResponseStoreStats
+        stockOnHand={data?.responseStoreStats.stockOnHand || 0}
+        incomingStock={data?.responseStoreStats.incomingStock || 0}
+        stockOnOrder={data?.responseStoreStats.stockOnOrder || 0}
+        requestedQuantity={data?.responseStoreStats.requestedQuantity || 0}
+        otherRequestedQuantity={data?.responseStoreStats.otherRequestedQuantity || 0}
+      />,
+      value: 'My Store',
```
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/DetailView/ResponseLineEdit/ResponseLineEdit.tsx:43]: can these be translations please?

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx (hunk 1 above)
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/DetailView/ReponseStats/ResponseStoreStats.tsx:81]:
```suggestion
      ? `${Math.round((100 * predictedStockLevels) / totalRequested).toString()}%`
```

**Review by mark-prins — APPROVED** [review]
👍  brilliant! that looks great, thanks 😁

File: client/packages/requisitions/src/ResponseRequisition/DetailView/ResponseLineEdit/ResponseLineEdit.tsx (hunk 9 above)
Comment (roxy-dao) [reply client/packages/requisitions/src/ResponseRequisition/DetailView/ResponseLineEdit/ResponseLineEdit.tsx:43]: Should the other tab values also be in translations as well then? (Details, Log)

**Pull request merged**